java/src/main/java/vw/VW.java

```diff
@@ -54,7 +47,44 @@ public static void main(String[] args) {
     public VW(String command) {
         isOpen = true;
         lock = new ReentrantLock();
-        nativePointer = initialize(command);
+        // This allows the user to instead of using the prepackaged JNI layer load their own external JNI layer.
```

Comment:
This comment should be a javadoc comment above the method so that people can refer to the API documentation rather than the code for an understanding of how to load their own natively compiled JNI extension.  I would also put this is a README.md so that people know it's possible.